includes/API_Credentials/API_Credentials_Manager.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ * Class WordPress\AI_Client\API_Credentials\API_Credentials_Manager
+ *
+ * @since n.e.x.t
+ * @package wp-ai-client
+ */
+
+namespace WordPress\AI_Client\API_Credentials;
+
+use WordPress\AiClient\AiClient;
+use WordPress\AiClient\Providers\DTO\ProviderMetadata;
+use WordPress\AiClient\Providers\Http\DTO\ApiKeyRequestAuthentication;
+
+/**
+ * Class for managing the AI API credentials for the various providers.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-import-type ProviderMetadataArrayShape from ProviderMetadata
+ */
+class API_Credentials_Manager {
+
+	private const OPTION_GROUP                = 'wp-ai-client-settings';
+	private const OPTION_PROVIDER_CREDENTIALS = 'wp_ai_client_provider_credentials';
+
+	/**
+	 * Initializes the API credentials manager.
+	 *
+	 * This method needs to be called by the consumer of this package, on the WordPress 'init' action hook.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function initialize(): void {
+		$this->collect_providers();
+		$this->register_settings();
+		$this->pass_credentials_to_client();
+
+		add_action(
+			'admin_menu',
+			function () {
+				$this->add_admin_screen();
+			}
+		);
```

Comment:
The closure seems unnecessary.
```suggestion
		add_action( 'admin_menu', array( $this, 'add_admin_screen' ) );
```

Reply:
The reason it's there is because the method is private. To me that makes more sense, because it shouldn't be called from the outside.

Reply:
Oh my goodness. 🤦‍♂️ 

Fun fact, I broke something with licensing in a product years ago because of this exact missed detail. 😅